manifests/mpm.pp

```diff
@@ -73,22 +73,31 @@
         }
       }
 
-      if $mpm == 'itk' and $::operatingsystem == 'Ubuntu' and $::operatingsystemrelease == '14.04' {
-        # workaround https://bugs.launchpad.net/ubuntu/+source/mpm-itk/+bug/1286882
-        exec {
-          '/usr/sbin/a2dismod mpm_event':
-            onlyif  => '/usr/bin/test -e /etc/apache2/mods-enabled/mpm_event.load',
-            require => Package['httpd'],
-            before  => Package['apache2-mpm-itk'],
-        }
-      }
-
       if $mpm == 'itk' and ( ( $::operatingsystem == 'Ubuntu' and $::operatingsystemrelease == '16.04' ) or ( $::operatingsystem == 'Debian' and versioncmp($::operatingsystemrelease, '9.0.0') >= 0 ) ) {
         $packagename = 'libapache2-mpm-itk'
       } else {
         $packagename = "apache2-mpm-${mpm}"
       }
 
+      $mod_enabled_dir = $::apache::mod_enable_dir
+
+      if $mpm == 'prefork' and ( $::operatingsystem == 'Debian' and versioncmp($::operatingsystemrelease, '9.0.0') >= 0 ) {
+        exec { '/usr/sbin/a2dismod mpm_event':
```

Comment:
Isn't this basically equivalent to a ```file``` with ```ensure => absent``` or does a2dismod do more than that? In that case you could even generalize it to ```if $mpm != 'event'``` instead of the OS checks.

Reply:
I assumed it did something else but it looks like all it does is remove the symlink in mods-enabled. 

I'm hesitant to generalize the conditional only because caution. However, I can't think of a situation where you wouldn't want to have event disabled if it wasn't the selected mpm. Further, maybe there should be logic to exclude any mpm that's not the selected one.

Reply:
> Further, maybe there should be logic to exclude any mpm that's not the selected one.

Heading to directory purging here :)

Reply:
Yes, that's basically symlink management.

The `a2*mod` commands actually were previously used by the apache module via the `a2mod` type (it's still there, deprecated and never removed; see https://github.com/puppetlabs/puppetlabs-apache/blob/3.0.0/lib/puppet/type/a2mod.rb). We tried writing providers to give similar functionality on several OSs, but it was very difficult to track all of the different changes across many OSs and each OS version.

The apache module was probably the first to specifically deviate from replicating OS defaults (RedHat-like on redhats and Debian-like on debian) and instead go for a one-size-fits-all. We did keep the mod symlink style via `apache::mod` but relied on recursive purging to keep everything inline inside the conf directory. This was 0.7.0 iirc.

What specifically causes the mpm module .load files to clash and not be purged? Is it a dependency between the mods and the file purging? Why do we have to exec out to `a2*mod` at all?